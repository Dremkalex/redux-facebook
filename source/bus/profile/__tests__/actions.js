// Actions
import { profileActions } from '../actions';

describe('profile actions', () => {
    test('fillProfile', () => {
        expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot();
    });

    test('update avatar', () => {
        expect(profileActions.updateAvatar(__.url)).toMatchSnapshot();
    });

    test('clear profile', () => {
        expect(profileActions.clearProfile()).toMatchSnapshot();
    });

    test('update name async', () => {
        expect(profileActions.updateNameAsync(__.newName)).toMatchSnapshot();
    });

    test('update avatar async', () => {
        expect(profileActions.updateAvatarAsync(__.newAvatar)).toMatchSnapshot();
    });

    test('update password async', () => {
        expect(profileActions.updatePasswordAsync(__.newPassword)).toMatchSnapshot();
    });
});
