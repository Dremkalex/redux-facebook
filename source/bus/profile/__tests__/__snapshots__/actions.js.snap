// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions clear profile 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`profile actions fillProfile 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`profile actions update avatar 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`profile actions update avatar async 1`] = `
Object {
  "payload": Array [
    "avatar",
  ],
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`profile actions update name async 1`] = `
Object {
  "payload": Object {
    "firstName": "Simon",
    "lastName": "Pirson",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`profile actions update password async 1`] = `
Object {
  "payload": Object {
    "NewPassword": "123456",
    "oldPassword": "12345",
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
